### PR TITLE
Fix set-env and make debugging OSX possible in DEBUG

### PR DESCRIPTION
**Pull request by premun**

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/Commands/XHarnessHelpCommand.cs [hunk 1]
```diff
@@ -50,8 +50,10 @@ public override int Invoke(IEnumerable<string> arguments)
                     PrintCommandHelp(new AndroidCommandSet(), subCommand);
                     break;
                 case "apple":
+#if !DEBUG
```
Comment (greenEkatherine) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/XHarnessHelpCommand.cs:53]: probably it should be `DEBUG` here?

Comment (premun) [reply src/Microsoft.DotNet.XHarness.CLI/Commands/XHarnessHelpCommand.cs:53]: We want to allow this command in DEBUG - so `#if RELEASE` require to be on OSX.

**Review by greenEkatherine — APPROVED** [review]

**Comment by imhameed** [comment]
I checked out xharness f4d7df9189d3b8ab04c6ba9a94dc8ccff255224e (which includes f4d7df9189d3b8ab04c6ba9a94dc8ccff255224e in its history) and built it. I'm no longer seeing the same failure I saw in https://github.com/dotnet/xharness/pull/496#issuecomment-806194987, but it doesn't look like the environment variable I passed to `--set-env` is being passed to the target app.

Here's the source code to the test app I'm using:
```objc
#import <UIKit/UIKit.h>
#import "AppDelegate.h"

#include <syslog.h>

int main(int argc, char * argv[]) {
    char const * what = getenv("MONO_APPLE_APP_ENTRY_POINT_LIB_NAME");
    what = what == NULL ? "<nullptr>" : what;
    syslog(LOG_USER | LOG_ERR, "XXXih: syslog: what = \"%s\"", what);
    printf("XXXih: stdout: what = \"%s\"\n", what);
    return 15;
}
```
All the extra "app stuff" was generated by XCode's 12.4's built-in template for an Objective-C iOS App using "Storyboard" and "UIKit App Delegate".

Here's what happens when I run the app in a simulator via xharness:
```
imhameed@ms-ih-mbp ~/Library/Developer/Xcode/DerivedData/mlaunchtestguy-fgmqnippyyklnifiecevtackjeai/Build/Products/Debug-iphonesimulator$ xharness.sh apple run --app=mlaunchtestguy.app --output-directory=out --targets=ios-simulator-64 --mlaunch=/Library/Frameworks/Xamarin.iOS.framework/Versions/14.10.0.4/bin/mlaunch --set-env=MONO_APPLE_APP_ENTRY_POINT_LIB_NAME=XXXihWhatWhat
XHarness command issued: apple run --app=mlaunchtestguy.app --output-directory=out --targets=ios-simulator-64 --mlaunch=/Library/Frameworks/Xamarin.iOS.framework/Versions/14.10.0.4/bin/mlaunch --set-env=MONO_APPLE_APP_ENTRY_POINT_LIB_NAME=XXXihWhatWhat
info: Preparing run for ios-simulator-64
info: Getting app bundle information..
info: Starting application 'mlaunchtestguy' on ios-simulator-64
info: App run ended with 15
fail: Application has finished with exit code 15 but 0 was expected
XHarness exit code: 71 (GENERAL_FAILURE)
```

Exit code 15 is expected. That's fine. But if I open the macOS console app and view the syslog output for xharness-created simulator, I'll see this:
```
XXXih: syslog: what = "<nullptr>"
```

So the environment variable isn't making it through. `out/run-ios-simulator-64.log` also contains this:
```
12:40:17.9236580 12:40:17.9236440 Xamarin.Hosting: The bundle id ihaaaaaaaaaaaaaaaa.mlaunchtestguy was successfully installed.
12:40:17.9515080 12:40:17.9514930 Xamarin.Hosting: Launching ihaaaaaaaaaaaaaaaa.mlaunchtestguy async on 'iOS 14.4 (18D46) - iPhone X' with: {
12:40:17.9515600 12:40:17.9515560     arguments =     (
12:40:17.9515720 12:40:17.9515700     );
12:40:17.9515820 12:40:17.9515800     environment =     {
12:40:17.9515910 12:40:17.9515890         NSUnbufferedIO = YES;
12:40:17.9515990 12:40:17.9515980         "OS_ACTIVITY_DT_MODE" = YES;
12:40:17.9516080 12:40:17.9516060     };
12:40:17.9516560 12:40:17.9516410 }
12:40:18.3217350 12:40:18.3217210 Xamarin.Hosting: Launched ihaaaaaaaaaaaaaaaa.mlaunchtestguy with pid 27793
12:40:19.3312950 12:40:19.3312820 Xamarin.Hosting: Simulated process has exited.
12:40:19.4933080 12:40:19.4932920 Process mlaunch exited with 0
```

**Comment by premun** [comment]
@imhameed my mistake! I was setting them for the mlaunch process instead of passing them to mlaunch. Fix is here: https://github.com/dotnet/xharness/pull/524

**Pull request merged**